### PR TITLE
Support if_exists and if_not_exists on create/drop table commands

### PR DESCRIPTION
**Pull request by agriffin-grow**
Fixes: #1520

### Description
Adds if_exists/if_not_exists support to table commands, mirroring closely the same logic for indexes.

### Checklist
This pull request is:

- [ ] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [x] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [ ] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

### CODE REVIEW

File: alembic/operations/base.py [hunk 1]
```diff
@@ -1247,6 +1251,10 @@ def create_table(
              quoting of the schema outside of the default behavior, use
              the SQLAlchemy construct
              :class:`~sqlalchemy.sql.elements.quoted_name`.
+            :param if_not_exists: If True, adds IF NOT EXISTS operator when
+             creating the table.
+
+             .. versionadded:: 1.13.4
```
Comment (agriffin-grow) [inline alembic/operations/base.py:1257]: Not really sure if this is the correct next version. Happy to change it.

Comment (CaselIT) [reply alembic/operations/base.py:1257]: it would be 1.13.3 since it's yet to be released

**Review by CaselIT — APPROVED** [review]
great work. It's just missing a changelog: see example in https://github.com/sqlalchemy/alembic/tree/main/docs/build/unreleased 

Feel free to add something like "pull request courtesy of <you>"

sorry for the delay in the review

**Comment by lachaib** [comment]
Hello,

My 2 cents on this PR, which I'm looking forward, is that it's missing the same thing I'm trying to fix with #1446 : we should have a small addition in `alembic/autogenerate/render.py` to render the option from autogen context.
It'd be great to have it from the start.

Thanks a lot for this pull request

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision 88c7b3fb9243bac57d0c91850214323bba200282 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change 88c7b3fb9243bac57d0c91850214323bba200282: https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/5455

**Review by zzzeek — CHANGES_REQUESTED** [review]
the tools/write_pyi.py script needs to be run here to generate appropriate stub files

**Comment by agriffin-grow** [comment]
@CaselIT @zzzeek updated based on your comments.

I also bumped against #1524 when using `tools/write_pyi.py` and put in a PR for that.

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision 1063a5ae1a8b9439415f112aaf46c6c65bb10be8 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset 1063a5ae1a8b9439415f112aaf46c6c65bb10be8 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/5455

**Comment by zzzeek** [comment]
hi -

still some failures here you may need to add skip markers to these tests for SQLAlchemy < 1.4

**Comment by agriffin-grow** [comment]
> hi -
> 
> still some failures here you may need to add skip markers to these tests for SQLAlchemy < 1.4

TY, I missed that tests still run for sqlalchemy 1.3 - should be updated now

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision 747ae3f345eca63f428140a2da5843ecc52f06e7 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset 747ae3f345eca63f428140a2da5843ecc52f06e7 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/5455

**Comment by sqla-tester** [comment]
**Michael Bayer** (zzzeek) wrote:

recheck

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/5455

File: tox.ini [hunk 2]
```diff
@@ -1,11 +1,11 @@
 [tox]
 
-
 envlist = py-sqlalchemy
 
 SQLA_REPO = {env:SQLA_REPO:git+https://github.com/sqlalchemy/sqlalchemy.git}
 
 [testenv]
+basepython = py311
```
Comment (sqla-tester) [inline tox.ini:8]: **Michael Bayer** (zzzeek) wrote:

ah.   this breaks things can we remove this please

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/5455

**Review by sqla-tester — COMMENTED** [review]
**Michael Bayer** (zzzeek) wrote:

code review left on gerrit

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/5455

File: tox.ini (hunk 2 above)
Comment (agriffin-grow) [reply tox.ini:8]: My bad, that shouldn't have been left in

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision ab4e7d4e535d9fd8fd56bb27e323bfdd864a7aec of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset ab4e7d4e535d9fd8fd56bb27e323bfdd864a7aec added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/5455

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision 469be01c6b5f9f42dc26017040a6fc54c4caef54 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset 469be01c6b5f9f42dc26017040a6fc54c4caef54 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/5455

**Review by sqla-tester — COMMENTED** [review]
**Michael Bayer** (zzzeek) wrote:

code review left on gerrit

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/5455

* tox.ini (line 8): Done

**Comment by sqla-tester** [comment]
**Federico Caselli** (CaselIT) wrote:

Great work!

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/5455

**Comment by agriffin-grow** [comment]
FYI I think @zzzeek might need to approve as well on account of the requested changes... but I also have no idea how Github and Gerrit interact with each other :)

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/5455 has been **merged**. Congratulations! :)

**Pull request closed without merging**